lib/active_merchant/billing/gateways/ixopay.rb

```diff
@@ -130,6 +134,22 @@ def add_debit(xml, money, options)
         end
       end
 
+      def add_preauth(xml, money, options)
+        currency    = options[:currency] || currency(money)
+        description = options[:description].blank? ? 'Preauthorize' : options[:description]
+
+        xml.preauthorize do
+          xml.transactionId new_transaction_id
+
+          add_customer_data(xml, options)
+
+          xml.amount      money
+          xml.currency    currency
```

Comment:
Optionally, to make this more concise, you could delete line 138 and replace this line with
`xml.currency options[:currency] || currency(money)`